safe/reportv4/extractors/analysis_detail.py

```diff
@@ -0,0 +1,206 @@
+# coding=utf-8
+from safe.definitionsv4.exposure import exposure_structure, exposure_road, \
+    exposure_land_cover, exposure_place
+from safe.definitionsv4.fields import exposure_type_field, \
+    exposure_class_field, hazard_count_field, total_affected_field, \
+    total_unaffected_field, total_field
+from safe.definitionsv4.hazard_classifications import all_hazard_classes
+from safe.reportv4.extractors.util import layer_definition_type
+from safe.utilities.i18n import tr
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = ':%H$'
+
+
+def analysis_detail_extractor(impact_report, component_metadata):
+    """
+    Extracting analysis result from the impact layer
+
+    :param impact_report: the impact report that acts as a proxy to fetch
+        all the data that extractor needed
+    :type impact_report: safe.reportv4.impact_report.ImpactReport
+
+    :param component_metadata: the component metadata. Used to obtain
+        information about the component we want to render
+    :type component_metadata: safe.reportv4.report_metadata.ReportMetadata
+
+    :return: context for rendering phase
+    :rtype: dict
+    """
+    context = {}
+
+    hazard_layer = impact_report.hazard_layer
+    exposure_layer = impact_report.exposure_layer
+    analysis_layer = impact_report.analysis_layer
+    analysis_layer_fields = analysis_layer.keywords['inasafe_fields']
+    analysis_feature = analysis_layer.getFeatures().next()
+    exposure_breakdown = impact_report.exposure_breakdown
+    exposure_breakdown_fields = exposure_breakdown.keywords['inasafe_fields']
+
+    """Initializations"""
+
+    # Get hazard classification
+    hazard_classification = None
+    # retrieve hazard classification from hazard layer
+    for classification in all_hazard_classes:
+        classification_name = hazard_layer.keywords['classification']
+        if classification_name == classification['key']:
+            hazard_classification = classification
+            break
+
+    # Get exposure type definition
+    exposure_type = layer_definition_type(exposure_layer)
+
+    # Analysis detail only applicable for:
+    applicable_types = [
```

Comment:
This is not driven by metadata. Please implement this in definitions package or fill this ticket : https://github.com/inasafe/inasafe/issues/3423